src/factory.js

```diff
@@ -193,6 +194,15 @@ module.exports = ({ Component, createElement, PropTypes }) => {
           ))
           const completeStateKeys = (this._stateKeys = keys(stateDescriptors))
 
+          this._resetState = () => {
+            const _initialstate = initialState(this.props)
+            keys(_initialstate).forEach(k => {
+              if (k in stateDescriptors) {
+                this._state[k] = _initialstate[k]
```

Comment:
Look at `setState` to see how to properly mutate the state.

We probably even should use it.